src/vm/eetwain.cpp

```diff
@@ -3099,7 +3099,17 @@ size_t EECodeManager::GetCallerSp( PREGDISPLAY  pRD )
     {
         EnsureCallerContextIsValid(pRD, NULL);
     }
-    return (size_t) (GetSP(pRD->pCallerContext));
+
+    size_t res = GetSP(pRD->pCallerContext);
+
+#ifdef _TARGET_X86_
+    EECodeInfo codeInfo;
+    codeInfo.Init((PCODE) pRD->ControlPC);
+
+    res -= codeInfo.GetCodeManager()->GetStackParameterSize(&codeInfo);
```

Comment:
@janvorli Then, CLR will get into stuck when resuming (please refer to 4th comment in #9366 for details).